react-native/react/tracker/index.js

```diff
@@ -1,89 +1,144 @@
 'use strict'
+/* @flow */
 
+// $FlowIssue base-react
 import React, {Component} from '../base-react'
+// $FlowIssue base-redux
 import {connect} from '../base-redux'
+// $FlowIssue platform dependent files
 import Render from './render'
-import {navigateUp} from '../actions/router'
+
+import * as trackerActions from '../actions/tracker'
+import { bindActionCreators } from 'redux'
+
+import type {UserInfo} from './bio.render.desktop'
+import type {Proof} from './proofs.render.desktop'
+import type {SimpleProofState} from '../constants/tracker'
+
+type TrackerProps = {
+  serverStarted: boolean,
+  proofState: SimpleProofState,
+  username: ?string,
+  shouldFollow: ?boolean,
+  reason: string,
+  userInfo: UserInfo,
+  proofs: Array<Proof>,
+  onCloseFromHeader: () => void,
+  onCloseFromActionBar: () => void,
+  onRefollow: () => void,
+  onUnfollow: () => void,
+  onFollowHelp: () => void,
+  onFollowChecked: () => void,
+  registerIdentifyUi: () => void
+}
 
 class Tracker extends Component {
-  constructor (props: any) {
-    super(props)
+  props: TrackerProps;
 
-    // this is TEMP since we don't have a store yet
-    this.state = {
-      shouldFollowChecked: props.shouldFollow
+  componentWillMount () {
+    if (!this.props.serverStarted) {
+      console.log('starting server')
+      this.props.registerIdentifyUi()
     }
   }
 
   render () {
     // these non-prop values will be removed during integration
-    return <Render {...this.props}
-      shouldFollow={this.state.shouldFollowChecked}
-      followChecked={checked => this.setState({shouldFollowChecked: checked})}
-      />
+    return <Render
+             bioProps={{
```

Comment:
if we give the flat props then Render would need to do this exact same thing. 

>  this component shouldn't know that level of detail

as far as this component is concerned it is simply fulfilling the type signature of what Render requires. It doesn't have or need any knowledge of what happens with these props, and the fact that they are called headerProps, actionProps, etc... are just a coincidence.

If we ever add/change a field in say proofs renderer, it's type signature changes, which would cause the `Render`er's type signature to change to and this wouldn't type check.

If we moved this to the render we'd have to do this assignment logic (example look how username is used in multiple places) in both desktop/mobile. Seems simpler to just have it here.